OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -410,6 +410,27 @@ public interface IRenderAboveShroudWhenSelected
 		bool SpatiallyPartitionable { get; }
 	}
 
+	public interface ISelection
+	{
+		int Hash { get; }
+		IEnumerable<Actor> Actors { get; }
+
+		void SetNotifySelection(IEnumerable<INotifySelection> worldNotifySelection);
+		void Add(Actor a);
+		void Remove(Actor a);
+		void OnOwnerChanged(Actor a, Player oldOwner, Player newOwner);
+		bool Contains(Actor a);
+		void Combine(World world, IEnumerable<Actor> newSelection, bool isCombine, bool isClick);
+		void Clear();
+		void Tick(World world);
```

Comment:
This can be removed from the interface by making `Selection` implement `ITick`.

This changes the behaviour slightly (we lose the guarantee about not changing the sync state), but this should not matter in practice.